### PR TITLE
Walled gardens v1 2

### PR DESCRIPTION
**Pull request by rejhgadellaa**
⚠️ This pull request is still in draft!

## Summary of Changes

* Update `walled-gardens.md` with v1.2 contents (WIP)
* Lots of minor tweaks (WIP)

### CODE REVIEW

**Comment by rejhgadellaa** [comment]
This PR also closes #36 and #35

(we may want to revisit those at some point to further improve on what I came up with here)

File: src/pages/walled-gardens.md
Comment (ssddanbrown) [inline src/pages/walled-gardens.md:2166]: The link in this paragraph has gone missing.

**Review by ssddanbrown — CHANGES_REQUESTED** [review]
Great effort on this, looks really good.

I've raised a bunch of comments, although I have been fairly picky so don't take them as required to merge if they'd cause frustration or time you don't have, merging this as-is is still a net benefit and many of my comments are opinion. 

Must admit, I haven't done a full comparison with the PDF, just a few spot-checks.
Also, I've did notice a few design/formatting tweaks & changes here and there which I haven't raised comments for, since they didn't cause issues and may just come down to opinion.

File: src/_includes/layouts/base.njk
Comment (rejhgadellaa) [reply src/_includes/layouts/base.njk:1]: `Math.random()` can end up returning a value we've already used, so that could cause people to get a cached version of the file. Data-based seemed safer as it will always increment.

File: src/_includes/partials/home-header.njk
Comment (rejhgadellaa) [reply src/_includes/partials/home-header.njk]: Ah that would be an unintentional change. I just wanted to make sure that we provide `width` and `height` attributes so we avoid layout shift. I forgot to add `height: auto` on the image, that's required for that trick to work:

```css
p.logo-home img {
  height: auto;
  vertical-align: bottom;
}
```

Addressed in fc7764e

As far as I can tell, that restores the old behaviour while keeping the attrs.

File: src/shortcodes/image.js
Comment (rejhgadellaa) [reply src/shortcodes/image.js]: Good catch. Fixed in 1286c82

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md]: Hmm Windows reports 7.7MB, but yep, good catch, thanks.

Fixed in 8f79fea

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:6]: Ah I did a search-and-replace on that because we were using both the with and without hyphen version.

Will double check with the team on what's correct.

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:181]: Hyphenation is intentional (see my earlier comment). The italic is something that was changed to bold in the 1.2 doc

image

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:2050]: Fixed in c33ee77

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:2056]: Fixed in c33ee77

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md]: Fixed in c33ee77

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md]: Fixed in c33ee77

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:2103]: Correct. Unfortunately, I don't know which header. I was able to fix most of them, but not all. This one actually slipped through the cracks, as I normally would just keep the `#heading-id` but not the entire url to the Google Doc :))

I'll keep this one open for now

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md]: Fixed in c33ee77

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:2559]: Yeah this is something I'd like to discuss... I think we need to make the images smaller and make them clickable. Thing is, a) do you/others agree, b) should we do that right away or is that something we can pick up after the v1.2 doc is live?

What do you think?

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:181]: Resolving this one as hyphenation is already mentioned (so I'll fix all of them in one go) and the bold is intentional

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:454]: Ah, I missed this comment. As mentioned in https://github.com/OpenWebAdvocacy/website/pull/33#discussion_r957650825, I agree but would like to find a solution like click-to-open/enlarge. The large images look a bit cluttered and many of them aren't very high quality to begin with...

We can also enlarge them a bit, 150px may be a bit too ~much~ little :)

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md]: Addressed in 5ff20c8

Note: Regarding the js, I've just moved the `pageNonCriticalScripts` front matter to `_includes/layout/paper.njk` as suggested above, so it works the same.

If we want to inline the script, I see two aproaches:
* just move the contents of `collapser.js` into a `<script>` tag in `paper.njk`, remove the `pageNonCriticalScripts` functionality
* Keep the `pageNonCriticalScripts` functionality but change the code in `base.njk` to `{% include %}` the js instead of linking to it

File: src/_includes/partials/home-header.njk
Comment (ssddanbrown) [reply src/_includes/partials/home-header.njk]: Prevention of layout shift is very welcomed, makes an especially positive difference on with the orange header banner.
Checking the preview now, things are looking aligned again, looks good!

File: src/pages/walled-gardens.md
Comment (ssddanbrown) [reply src/pages/walled-gardens.md:2413]: I think this really comes down to preference. Personally, I like them being readable by default as it avoids the extra required interaction to read them, but I don't feel too strongly either way.

> is that something we can pick up after the v1.2 doc is live?

Sure, not critical so not something that should block progression.

**Review by ssddanbrown — APPROVED** [review]
I've responded to, or resolved, any remaining points I could.
Is looking good so I'll approve for merge but I'll leave it up to you if you'd first want to address those final minor points before merge.

File: src/pages/walled-gardens.md
Comment (rejhgadellaa) [reply src/pages/walled-gardens.md:2413]: I'll create a separate issue for this. We can:

- Make the images a bit larger. I've currently set `max-height: 320px`, we could increase that a bit or maybe base it on `vh`?
- Make them clickable for full size. I'd like to automate so we get `<a />` for all gallery/screenshot images. Easiest would be to do it client side of course, but I can probably hack together a 11ty plugin that takes care of it.

And we can do both, of course! :)

**Pull request merged**
